### DISCUSSION TITLE
Opt-in to allow writable tokens on Dependabot workflows

### DISCUSSION THREAD

**Discussion by asciimike** [community/community#26805] [Actions]
<p>Currently, Actions that run off Dependabot PRs don’t have writable tokens, as they are treated like PRs from forks (c.f.: <a href="https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/" class="inline-onebox" rel="noopener nofollow ugc">GitHub Actions: Workflows triggered by Dependabot PRs will run with read-only permissions | GitHub Changelog</a>).</p>
<p>Dependabot users in <a href="https://github.com/dependabot/dependabot-core/issues/3253" class="inline-onebox" rel="noopener nofollow ugc">Dependabot cant read secrets anymore · Issue #3253 · dependabot/dependabot-core · GitHub</a> have a desire to opt-in events to allow Dependabot triggered workflows to have writable tokens.</p>
<p>As an example:</p>
<pre><code class="lang-auto">name: Dependabot Workflow
on:
  pull_request

# allow permissions to be up-scoped
permissions:
  pull_requests: write

jobs:
  do-stuff:
    runs-on: ubuntu-latest
    if: ${{ github.actor == 'dependabot[bot]' }}
    steps:
      - uses: ...
</code></pre>

**Comment by Simran-B** [comment]
<p>Should this topic be pinned? If I recall correctly, then there were quite a few other topics around this topic.</p>
<p>BTW: Does this also help in the following scenario?</p>
<ul>
<li>I have a repository <code>main</code> and a repository <code>other</code> in the same org</li>
<li>A workflow is supposed to run in <code>main</code> and push a new branch to <code>other</code>
</li>
<li>By default, the push to the other repo fails because of lack of permissions. I could use a PAT, but it would be strange to use one for an automation like this that should be attributed to the github actions bot.</li>
</ul>

**Comment by asciimike** [comment, marked as answer]
<p>The Actions team has shipped this feature: <a href="https://github.blog/changelog/2021-10-06-github-actions-workflows-triggered-by-dependabot-prs-will-respect-permissions-key-in-workflows/" class="inline-onebox">GitHub Actions: Workflows triggered by Dependabot PRs will respect permissions key in workflows | GitHub Changelog</a></p>
